### PR TITLE
Pfappserver Updates 

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Delete branch after merge
YES

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/components/DeleteButton.vue [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
       >{{ confirm }}</b-button>
-    <b-button 
-      v-if="!interrupt" 
-      type="button" 
-      :variant="variant" 
-      :disabled="disabled"
-      @click="startInterrupt($event)"
-      ><slot>{{ $t('Delete') }}</slot></b-button>
+    <!-- RTL END -->
+
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/DeleteButton.vue]: Suggestion: handle the direction through CSS [flex-direction](https://css-tricks.com/almanac/properties/f/flex-direction/).

File: html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue [hunk 2]
```diff
@@ -185,6 +200,56 @@ export default {
...
+          text = text.slice(0, -1)
+        }
+        if (amount < 0) {
+          // make absolute
+          amount = 0 - amount
+          text += ' ago'
+        } else {
+          text += ' from now'
+        }
+        return amount.toString() + ' ' + this.$i18n.t(text)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfFormDatetime.vue]: Works great in English, less in French.

> *3 hours ago*
> *il y a 3 heures*

We should use a [named or list literal](https://kazupon.github.io/vue-i18n/guide/formatting.html#html-formatting) or rely on timeago/date-fns.

File: html/pfappserver/root/static.alt/src/components/pfFormPrefixMultiplier.vue [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  <b-form-group :label-cols="labelCols" :label="$t(label)" :state="isValid()" :invalid-feedback="$t(invalidFeedback)" class="mb-0" horizontal>
+    <b-input-group>
+      <b-input-group-prepend v-if="prependText" is-text>
+        {{ prependText }}
+      </b-input-group-prepend>
+      <b-form-input type="number" :placeholder="placeholder" v-model="inputValue" @input.native="validate()" :state="isValid()"></b-form-input>
+      <b-form-text v-if="text" v-t="text"></b-form-text>
+      <b-input-group-append>
+        <b-button-group v-if="prefixes.length > 0" rel="prefixButtonGroup">
+          <b-button v-for="(prefix, index) in prefixes" v-if="inRange(index)" :key="index" :variant="[prefix.selected ? 'primary' : 'light']" v-b-tooltip.hover.bottom.d300 :title="$t(prefix.name + units.name)" @click.stop="changeMultiplier($event, index)">{{ prefix.label + units.label }}</b-button>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfFormPrefixMultiplier.vue]: The button title should be localized.

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 4]
```diff
@@ -195,12 +198,14 @@ export default {
...
       const _this = this
       this.$store.dispatch(`${this.storeName}/addSavedSearch`, {name: this.saveSearchString, query: this.condition}).then(response => {
+        _this.$store.dispatch('notification/info', {message: _this.$i18n.t('Search saved as ') + '\'' + _this.saveSearchString + '\''})
         _this.saveSearchString = ''
         _this.showSaveSearchModal = false
       })
     }
   },
   mounted () {
+    console.log(['pfSearch storeName', this.storeName])
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue]: Remove this debugging output.

File: html/pfappserver/root/static.alt/src/globals/pfRegExp.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ *  This is not meant to be a validation replacement, but rather a
+ *    helper that actively assists the user during form input.
+ *
+ *  See the `filter` property in:
+ *    ../components/pfFormInput.vue
+**/
+export const pfRegExp = {
+  none: null,
+  integer: /^[-0-9]*$/,
+  integerAbsolute: /^[0-9]*$/,
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/pfRegExp.js]: I prefer `positiveInteger`. What do you think?

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue [hunk 6]
```diff
@@ -5,6 +5,7 @@
       <b-card-header>
         <b-button-close @click="close" v-b-tooltip.hover.left.d300 :title="$t('Close [ESC]')"><icon name="times"></icon></b-button-close>
         <h4 class="mb-0">MAC <strong v-text="mac"></strong></h4>
+        ---{{ testName }}---
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: Debugging leftover?

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue [hunk 7]
```diff
@@ -473,6 +492,9 @@ export default {
     close () {
       this.$router.push({ name: 'nodes' })
     },
+    reset () {
+      // noop
+    },
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: Where is this call? Should it be removed?

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesCreate.vue [hunk 8]
```diff
@@ -192,8 +196,13 @@ export default {
...
       if (this.modeIndex === 0) {
         this.$store.dispatch('$_nodes/createNode', this.single).then(response => {
-          console.debug('node created')
+          this.$store.dispatch('notification/info', {message: this.$i18n.t('Node') + ' ' + this.single.mac + ' ' + this.$i18n.t('created')})
+          this.single = {
+            mac: '',
+            status: 'reg'
+          }
         }).catch(err => {
+          this.$store.dispatch('notification/danger', {message: this.$store.state.$_nodes.message})
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesCreate.vue]: I guess we should now use `this.storeName` everywhere.

**Review by cgx — CHANGES_REQUESTED** [review]

File: html/pfappserver/root/static.alt/src/views/Nodes/_router/index.js [hunk 9]
```diff
@@ -24,7 +24,7 @@ const route = {
     {
       path: 'search',
       component: NodesSearch,
-      props: (route) => ({ query: route.query.query, storeName: '$_nodes' })
+      props: (route) => ({ storeName: '$_nodes' query: route.query.query })
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_router/index.js]: Looks like a comma is missing.

File: html/pfappserver/root/static.alt/src/views/Nodes/_router/index.js [hunk 10]
```diff
@@ -24,7 +24,7 @@ const route = {
     {
       path: 'search',
       component: NodesSearch,
-      props: (route) => ({ storeName: '$_nodes' query: route.query.query })
+      props: (route) => ({ storeName: '$_nodes', query: route.query.query })
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_router/index.js]: Forget my previous comment :-)

**Comment by satkunas** [comment]
@cgx ready for re-re-view

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesCreate.vue [hunk 11]
```diff
@@ -202,7 +202,7 @@ export default {
             status: 'reg'
           }
         }).catch(err => {
-          this.$store.dispatch('notification/danger', {message: this.$store.state.$_nodes.message})
+          this.$store.dispatch('notification/danger', {message: this.$store.state[this.storeName].message})
           console.debug(err)
           console.debug(this.$store.state.$_nodes.message)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesCreate.vue]: Also use `this.storeName` here.

Rebase your branch and we'll be good to merge!

**Review by cgx — APPROVED** [review]

**Pull request merged**